parcels/field.py

```diff
@@ -310,8 +310,13 @@ def eval(self, time, x, y):
         return self.units.to_target(value, x, y)
 
     def ccode_eval(self, var, t, x, y):
-        return "temporal_interpolation_linear(%s, %s, %s, %s, %s, %s, &%s)" \
-            % (x, y, "particle->xi", "particle->yi", t, self.name, var)
+        # Casting interp_method to int as easier to pass on in C-code
+        if self.interp_method is 'linear':
+            interp_method = 0
```

Comment:
Left it as is now. I tried creating a class `InterpCode` just like in kernel.error
```
class InterpCode(IntEnum):
    Linear = 0
    Nearest = 1
```
 but not sure how to implement then in `ccode_eval`. When I try
```
    def ccode_eval(self, var, t, x, y):
        # Casting interp_method to int as easier to pass on in C-code
        if self.interp_method is 'linear':
            interp_method = InterpCode.Linear
        elif self.interp_method is 'nearest':
            interp_method = InterpCode.Nearest
        return "temporal_interpolation_linear(%s, %s, %s, %s, %s, %s, &%s, %s)" \
            % (x, y, "particle->xi", "particle->yi", t, self.name, var, interp_method)
```
I get a compilation error
```
error: unexpected type name 'InterpCode': expected expression
    err = temporal_interpolation_linear(particle->lon, particle->lat, particle->xi, particle->yi, time, P, &tmp0, InterpCode.Nearest);
                                                                                                                  ^
```

Am I missing something?